src/controllers/scenario/index.ts

```diff
@@ -43,15 +45,19 @@ ${eof}`
     return new Promise((resolve) => {
       exec(`${scp}; ${ssh}`, (err, stdout) => {
         exec(rm) // fire and forget, remove binary file
-        return resolve({
-          status: err ? 'error': 'success',
+        return resolve(err ? {
+          status: 'error',
+          output: stdout,
+          exception: err
+        } : {
+          status: 'success',
           output: stdout,
-          ...err ? { exception: err } : {}
         })
       })
     })
   }
 
+  @Response<ValidateErrorJSON>(422, "Validation Failed")
```

Comment:
Yes! This is wahat I wanted, but have not checked any docs or anything regarding this

Reply:
Wonder how much we can expand on this e.g. getting error instance to return some nice formatted stuff back to the client.

Reply:
What response looks like? of this 422

Reply:
It's actually more yucky than I'd like for example wit ha bad executable:

```json
{
  "message": "Validation Failed",
  "details": {
    "executable": {
      "message": "Could not match the union against any of the items. Issues: [{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-access-aarch64']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-access-cheri']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-read-aarch64']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-read-cheri']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-readV2-aarch64']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-readV2-cheri']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-write-aarch64']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-write-cheri']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['use-after-free-aarch64']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['use-after-free-cheri']\",\"value\":\"ou-of-bounds-read-aarch64\"}}]",
      "value": "ou-of-bounds-read-aarch64"
    }
  }
}
```

But unfortunately that's what tsoa produces for the validation error. My implementation btw is modified from https://tsoa-community.github.io/docs/error-handling.html#setting-up-error-handling